app/submission/workflow/metadata/route.js

```diff
@@ -1,4 +1,37 @@
 import Route from '@ember/routing/route';
+import { get } from '@ember/object';
 
 export default Route.extend({
+  async model() {
+    let submission = await this.modelFor('submission.workflow').submission;
+    let metadata = await get(submission, 'metadataModel');
+
+    if (metadata) return metadata;
+
+    let authors = [
+      await this.store.createRecord('author'),
+    ];
+
+    let grants = [
+      await this.store.createRecord('grant'),
+    ];
+
+    let publications = [
+      await this.store.createRecord('publication'),
+    ];
+
+    let contacts = [
+      await this.store.createRecord('contact'),
+    ];
+
+    metadata = await this.store.createRecord('metadata', {
+      authors,
+      grants,
+      publications,
+      contacts,
+      submission,
+    });
+
+    return metadata;
```

Comment:
Just a comment, would it be worth RSVP.hash or even ember-concurrency stuff?

I think something like this should functionally be the same as the sequential `await`s
```
    return hash({
      authors: A([this.store.createRecord('author')]),
      grants: A([this.store.createRecord('grant')]),
      publications: A([this.store.createRecord('publication')]),
      contacts: A([this.store.createRecord('contact')]),
      submission
    });
```

Reply:
After thinking about this one a bit more I think it'd be best to let the model hook return the primary metadata model and allow the relationships provide access to side models. I usually reserve returning a hash for where there is some async behavior that makes it preferable to guarantee data is loaded by the time rendering occurs. Here, I think the fetching of the related models will occur pretty quickly and it complicates the structure of things down in the component if I move to a hash here.

I did upgrade to ember arrays though, that seems like a decent improvement.

Lately, I've been using async more in routes than ember-concurrency just because we don't have any tear down concerns. If something needs to happen that's specialized I'll reach for e-c, but I think here it may be okay to leave it.